examples/vae/train_vae.py

```diff
@@ -20,60 +21,79 @@ def main():
     parser.add_argument('--test', action='store_true',
                         help='Use tiny datasets for quick tests')
     args = parser.parse_args()
 
     print('GPU: {}'.format(args.gpu))
-    print('# dim z: {}'.format(args.dimz))
-    print('# Minibatch-size: {}'.format(args.batchsize))
+    print('# dim z: {}'.format(args.dim_z))
+    print('# Minibatch-size: {}'.format(args.batch_size))
     print('# epoch: {}'.format(args.epoch))
     print('')
 
     # Prepare VAE model, defined in net.py
-    model = net.VAE(784, args.dimz, 500)
+    encoder = net.make_encoder(784, args.dim_z, args.dim_h)
+    decoder = net.make_decoder(784, args.dim_z, args.dim_h)
+    prior = net.make_prior(args.dim_z, device=args.gpu)
+    avg_elbo_loss = net.AvgELBOLoss(encoder, decoder, prior,
+                                    beta=args.beta, k=args.k)
 
     # Setup an optimizer
     optimizer = chainer.optimizers.Adam()
-    optimizer.setup(model)
+    optimizer.setup(avg_elbo_loss)
 
     # Initialize
     if args.initmodel:
-        chainer.serializers.load_npz(args.initmodel, model)
+        chainer.serializers.load_npz(args.initmodel, avg_elbo_loss)
 
     # Load the MNIST dataset
     train, test = chainer.datasets.get_mnist(withlabel=False)
+
+    # Binarize dataset
+    train[train >= 0.5] = 1.0
+    train[train < 0.5] = 0.0
+    test[test >= 0.5] = 1.0
+    test[test < 0.5] = 0.0
+
     if args.test:
         train, _ = chainer.datasets.split_dataset(train, 100)
         test, _ = chainer.datasets.split_dataset(test, 100)
 
-    train_iter = chainer.iterators.SerialIterator(train, args.batchsize)
-    test_iter = chainer.iterators.SerialIterator(test, args.batchsize,
+    train_iter = chainer.iterators.SerialIterator(train, args.batch_size)
+    test_iter = chainer.iterators.SerialIterator(test, args.batch_size,
                                                  repeat=False, shuffle=False)
 
     # Set up an updater. StandardUpdater can explicitly specify a loss function
     # used in the training with 'loss_func' option
     updater = training.updaters.StandardUpdater(
         train_iter, optimizer,
-        device=args.gpu, loss_func=model.get_loss_func())
+        device=args.gpu, loss_func=avg_elbo_loss)
 
     trainer = training.Trainer(updater, (args.epoch, 'epoch'), out=args.out)
-    trainer.extend(extensions.Evaluator(test_iter, model, device=args.gpu,
-                                        eval_func=model.get_loss_func(k=10)))
+    trainer.extend(extensions.Evaluator(
+        test_iter, avg_elbo_loss, device=args.gpu))
     trainer.extend(extensions.dump_graph('main/loss'))
     trainer.extend(extensions.snapshot(), trigger=(args.epoch, 'epoch'))
     trainer.extend(extensions.LogReport())
     trainer.extend(extensions.PrintReport(
         ['epoch', 'main/loss', 'validation/main/loss',
-         'main/rec_loss', 'validation/main/rec_loss', 'elapsed_time']))
+         'main/rec', 'main/penalty', 'elapsed_time']))
```

Comment:
`rec` might be too short to guess its meaning.
I'd write `reconstr` and `kl_div`.  What do you think?

Reply:
Thank you for your recommendation. I think `kl_div` is still vague because this name does not specify KL divergence for what. Since I thought that the variable name of KL penalty should imply that it is the penalty term of the loss function, I have changed `rec` and `penalty` to `reconstr` and `kl_penalty`: add8bfe